### DISCUSSION TITLE
ExtAddress changes

### DISCUSSION THREAD

**Discussion by dakhnod** [openthread/openthread#10874] [Q&A]
When running `ot-ctl extaddr`, the cli reports some Address, which mostly stays persistent.
When I use the same RCP with another computer, or sometimes on the same computer, the ExtAddr changes.

Here are my questions:
1. When is the ExtAddr generated?
2. Under what circumstances does it change?
3. Besides the ExtAddr, how can I uniquely identify an RCP?
4. How can I uniquely identify a Border router?

**Comment by jwhui** [comment]
> 1. When is the ExtAddr generated?

Thread randomly generates a new Extended Address when the interface is commissioned.

> 2. Under what circumstances does it change?

See above.

> 3. Besides the ExtAddr, how can I uniquely identify an RCP?

You can try observing the EUI-64 value, which is a factory-provisioned unique identifier.

> 4. How can I uniquely identify a Border router?

How to uniquely identify a device is product specific. One possibility is by using the EUI-64.

**Reply by xiaojiang2017** [reply]
Hi @jwhui 
Is there any way to know the mapping between EUI64 and Extended Address when the Joiner is commissioned?
The `meshdiag` command can only obtain Extended Address.

**Reply by jwhui** [reply]
The Joiner generates a new random Extended Address after successfully completing the commissioning process:
https://github.com/openthread/openthread/blob/d3d91634f465822b3a4c23da1347efc9a47f2de4/src/core/meshcop/joiner.cpp#L572

The device will need to report its EUI-64 and Extended Address after to properly map them.

**Comment by dakhnod** [comment]
@jwhui Thanks, great! Is there a way to discover the EUI64 via CoAP diagnostics or similar?

**Reply by jwhui** [reply]
Yes, see: https://github.com/openthread/openthread/blob/cb27207675b2e92119a9ba5eb4b2f366be0852c7/include/openthread/netdiag.h#L67

**Reply by dakhnod** [reply]
ah, perfect! thanks!
